module/modmanager/manager.go

```diff
@@ -225,6 +225,7 @@ func (mgr *Manager) Add(ctx context.Context, confs ...config.Module) error {
 		go func(i int, conf config.Module) {
 			defer wg.Done()
 
+			mgr.logger.CInfow(ctx, "now adding module", "module", conf.Name)
 			err := mgr.add(ctx, conf)
```

Comment:
It seems `add` can behave such that:
* A process is not started
* Nor is an error log line not produced.

This is due [to this early](https://github.com/viamrobotics/rdk/pull/3716/files#diff-a4308b77f384da81096eda21bf023471e80818dda86e74f9335f99a6a57d07c9R243-R246) return.

Do we know whether/why that can happen? If not, let's add another log line for that case.

Reply:
We can definitely early return there if we attempt to add a module that already exists in the module manager (e.g. when partial start is enabled and a duplicate module is added in a separate batch from its existing duplicate). Added a log for transparency.